### DISCUSSION TITLE
Initialise atom as null

### DISCUSSION THREAD

**Discussion by philipaarseth** [pmndrs/jotai#629] [Q&A] [answered]
If you initialise an atom like this

`export const objState = atom<Obj | null>(null)`

It will think I want a read only atom and using it like a normal read/write atom will cause type issues as 
`const [obj, setObj] = useAtom(objState)`
setObj will be of type never and calling it will make typescript complain never has no call signatures.

However if I use any other type than undefined or null like for instance `atom<Obj | number>(0)`
I can do `if(typeof raid === "number") return` just fine.

So how does one initialise an atom as null, then later check if that value is null and use that as a bail out mechanic?
Alternatively if there's another better approach here I'm all ears!

**Comment by dai-shi** [comment, marked as answer]
A quick question: Do you have `"strict": true` in tsconfig.json?

**Reply by philipaarseth** [reply]
No, `"strict": false,`

**Reply by dai-shi** [reply]
Can you try with `true`? It can't infer `null` properly with `false`.

Please see https://github.com/pmndrs/jotai/issues/550#issuecomment-877777563 for a workaround with `"strict": false`.

**Reply by philipaarseth** [reply]
Well it looks to be working, typescript isn't complaining about that part anymore.
However, I don't know if I have time to rewrite the project to be strict mode, also I'm not a huge fan of all the boilerplate. 

Is there no other way?

**Reply by dai-shi** [reply]
To my knowledge, it's a limitation in TS, and I don't know any workaround other than that.
If your code allows it, you can use `false` instead of `null`. `const objState = atom<Obj | false>(false)`

**Reply by philipaarseth** [reply]
That's a shame, but alright I think that will have to do for now.
Thanks a ton!
